src/Microsoft.Dnx.Watcher.Core/ProjectWatcher.cs

```diff
@@ -0,0 +1,286 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.Versioning;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Dnx.Compilation;
+using Microsoft.Dnx.Runtime;
+using Microsoft.Dnx.Runtime.Common.Impl;
+using Microsoft.Framework.Logging;
+
+namespace Microsoft.Dnx.Watcher.Core
+{
+    public class ProjectWatcher
+    {
+        private readonly Func<string, IFileWatcher> _fileWatcherFactory;
+        private readonly IProjectGraphProvider _projectGraphProvider;
+
+        private readonly ILoggerFactory _loggerFactory;
+        private readonly ILogger _logger;
+
+        private string _fullProjectFilePath;
+        private string _rootPath;
+
+        public ProjectWatcher(
+            Func<string, IFileWatcher> fileWatcherFactory,
```

Comment:
This is for now because you wanted the same behavior as the current `--watch`. Once I remove the old behavior I can take a dependency on a better file watcher